### ISSUE TITLE
why didn't UserCountry.getCountry, UserCountry.getContinent, UserCountry.getRegion statistics?

### ISSUE THREAD

**Issue by lixinxiaobai** [piwik/piwik#10048]
Why didn't UserCountry.getCountry, UserCountry.getContinent, UserCountry.getRegion statistics?
only UserCountry.getNumberOfDistinctCountries?
For example: 
UserCountry.getNumberOfDistinctContinents,
UserCountry.getNumberOfDistinctgetRegions are not these API

**Comment by tsteur** [comment]
There's probably no particular reason for it. Just that it wasn't needed so far. 

I don't quite understand which API's you need exactly? Do I get it right you want `UserCountry.getNumberOfDistinctContinents` and `UserCountry. getNumberOfDistinctRegions`? You could probably always request all Continents or all regions with `filter_limit=-1` and then count the number of entries in the result.

**Comment by lixinxiaobai** [comment]
Right, I am get all continents or city, then calculates the number of items to obtain the total number of, but such speed is not good.

So I want to have a method like `UserCountry.getNumberOfDistinctCountries`  can be obtained directly from the count

**Comment by tsteur** [comment]
True this is surely not perfect. I will adjust the title of the issue.

I recently thought of actually deprecating the `getNumberOfDistinctCountries ` method at some point but on the other side we could also add more such API methods also for other APIs. 

When we work on the new archiver in https://github.com/piwik/piwik/issues/7470 we could actually even archive such metrics automatically for all reports, and eventually maybe even add such API methods automatically as well.

**Comment by mattab** [comment]
> When we work on the new archiver in #7470 we could actually even archive such metrics automatically for all reports, and eventually maybe even add such API methods automatically as well.

+1

Such report metadata could be returned by the [Metadata API reports](http://developer.piwik.org/api-reference/reporting-api-metadata) yet without requiring to load the full processed report.
